File: extensions/wee.py
cmd = "wee %1"
name = "Weeeeeeeeeeeeeeeeeeeeeeeeeeeeee"
filext = "*.wee *.w"


def syntaxhighlighter(text):
    colors = {
        "Start epidemic": "#0D66AB",
        "Infect person": "#007F00",
        "Deinfect person": "#0D17AB",
        "Bulk infect": "#0DAD51",
        "Check number of infections": "#414DF1",
        "Bulk deinfect": "#0DABA0",
        "Skip next if no one infected": "#41A5F1",
        "Delevop vaccine": "#11E4D6",
    }
    result = []
    for i in text.split("\n")[:-1]:
        if i in colors:
            result += [(colors[i], i)]
        else:
            result += [("#000000", i)]
        result += [("#000000", "\n")]
    if text.split("\n")[-1] in colors:
        result += [(colors[text.split("\n")[-1]], text.split("\n")[-1])]
    else:
        result += [("#000000", text.split("\n")[-1])]
    return result

File: bin/wee.cpp
#include <iostream>
#include <string>
#include <fstream>
std::string cmd;
char accumulator;
int main(int argc,char *argv[])
{
    std::ifstream f(argv[1]);
    std::getline(f, cmd);
    if (cmd != "Start epidemic") {
        std::cerr << "Program must start with Start epidemic";
        return 1;
    }
    while (cmd!="Delevop vaccine") {
        if (f.eof()) continue;
        std::getline(f, cmd);
        if (cmd == "Infect person") ++accumulator;
        if (cmd == "Deinfect person") --accumulator;
        if (cmd == "Bulk infect") accumulator = std::cin.get();
        if (cmd == "Bulk deinfect") accumulator = bool(accumulator);
        if (cmd == "Check number of infections") putchar(accumulator);
        if (cmd == "Skip next if no one infected" && !accumulator) std::getline(f, cmd),cmd="";
    }
    return 0;
}
